tutorials/firstcodemod.md

```diff
@@ -155,6 +155,14 @@ namespace Celeste.Mod.Example {
 }
 ```
 
+### Hooking Methods
+There are currently 3 ways to hook methods. These are:
+
+* RuntimeDetour: This is currently used by [RainbowMod](https://github.com/EverestAPI/RainbowMod) and [MadelineEnergySelector](https://github.com/EverestAPI/MadelineEnergySelector), and will be deprecated for end-user mods once HookedMethod has a stable version.
+* HookedMethod.Hook: This is the replacement for RuntimeDetour, intended to be more user-friendly and have more features. Currently, it has nothing except a slightly simpler interface, and a backbone to easily allow adding IL-manipulation and complex non-detour functionality. RainbowMod will most likely be ported to HookedMethod.Hook in the near future. This is currently recommended for new mods that do not require a stable development interface. Documentation is available in [`Examples/Examples.cs`](https://github.com/EverestAPI/HookedMethod/blob/master/Examples/Program.cs) in the HookedMethod repo.
+* Everest.Events: This is the hooking method built in to Everest, and is used by [GhostMod](https://github.com/EverestAPI/GhostMod).
+  * There's an early WIP to replace this with a feature of HookedMethod, HookedMethod.EventHook. MadelineEnergySelector may be ported to use this instead of RuntimeDetour.
```

Comment:
I'm confused, I thought we'd offer _both_ `Everest.Events` _and_ hook events, keeping former to expose general-purpose events in a newcomer-friendly manner, instead of throwing newcomers into the cold water and telling them to "use ILSpy?"

Reply:
I thought what we'd decided on was to use no-op extension methods created by Everest that get called in place of events being triggered, and either hack together Intellisense via the relinker, or make a plugin for VS and VSCode to display Everest events. If we go the plugin route, we could possibly use the Omnisharp database(?) to just use the same autocomplete as a non-EventHook method reference.

Reply:
Hmm, seems like there's still some confusion on that front 😅 maybe we should clear this up in Discord?

I'm all in for a plugin-less modding environment.

Reply:
👍 to replace `Everest.Events` with autogenerated `HookedEvents` + manually maintained, "guided" `HookedEvents` that map to the existing `Everest.Events`